src/wpt/harness.ts

```diff
@@ -32,12 +32,68 @@ import {
   throws,
 } from 'node:assert';
 
-function OptionalFeatureUnsupportedError(message) {
-  AssertionError.call(this, message);
+type Options = {
+  expectedFailures?: string[];
+  verbose?: boolean;
+  skippedTests?: string[];
+};
+
+type TestRunnerFn = (callback: TestFn | PromiseTestFn, message: string) => void;
+type TestFn = () => void;
+type PromiseTestFn = () => Promise<void>;
+type ThrowingFn = () => any;
+
+declare namespace globalThis {
+  var errors: Error[];
+  var testOptions: Options;
+  var Window: typeof globalThis;
+  var self: typeof globalThis;
+  var GLOBAL: { isWindow(): boolean };
+  function fetch(url: string): Promise<{ json(): Promise<any> }>;
+  function test(func: TestFn, name: string): void;
+  function done(): undefined;
+  function subsetTestByKey(
+    _key: any,
+    testType: TestRunnerFn,
+    testCallback: TestFn | PromiseTestFn,
+    testMessage: string
+  ): void;
+  function promise_test(
+    func: PromiseTestFn,
+    name: string,
+    properties?: any
+  ): void;
+  function assert_equals(a: any, b: any, message?: string): void;
+  function assert_not_equals(a: any, b: any, message?: string): void;
+  function assert_true(val: any, message?: string): void;
+  function assert_false(val: any, message?: string): void;
+  function assert_array_equals(a: any, b: any, message?: string): void;
+  function assert_object_equals(a: any, b: any, message?: string): void;
+  function assert_implements(condition: any, description?: string): void;
+  function assert_implements_optional(
+    condition: any,
+    description?: string
+  ): void;
+  function assert_unreached(description?: string): void;
+  function assert_throws_js(
+    constructor: any,
+    func: ThrowingFn,
+    description?: string
+  ): void;
+  function assert_throws_exactly(
+    exception: any,
+    fn: ThrowingFn,
+    description?: string
+  ): void;
+  function assert_throws_dom(
+    type: any,
+    funcOrConstructor: any,
+    descriptionOrFunc: any,
+    maybeDescription: any
+  ): void;
 }
-OptionalFeatureUnsupportedError.prototype = Object.create(
```

Comment:
Let's just add a ts-expect-error here and keep it as it is. I took the code from the actual wpt harness script in web-platform-tests repository.